github/interactions_orgs.go

```diff
@@ -0,0 +1,73 @@
+// Copyright 2018 The go-github AUTHORS. All rights reserved.
+//
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package github
+
+import (
+	"context"
+	"fmt"
+)
+
+// GetRestrictionsForOrg fetches the interaction restrictions for an organization.
+//
+// GitHub API docs: https://developer.github.com/v3/interactions/orgs/#get-interaction-restrictions-for-an-organization
+func (s *InteractionsService) GetRestrictionsForOrg(ctx context.Context, organization string) (*InteractionRestriction, *Response, error) {
+	u := fmt.Sprintf("orgs/%v/interaction-limits", organization)
+	req, err := s.client.NewRequest("GET", u, nil)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	// TODO: remove custom Accept header when this API fully launches.
+	req.Header.Set("Accept", mediaTypeInteractionRestrictionsPreview)
+
+	organizationInteractions := new(InteractionRestriction)
+
+	resp, err := s.client.Do(ctx, req, organizationInteractions)
+	if err != nil {
+		return nil, resp, err
+	}
+
+	return organizationInteractions, resp, nil
+}
+
+// UpdateRestrictionsForOrg adds or updates the interaction restrictions for an organization.
+//
+// GitHub API docs: https://developer.github.com/v3/interactions/orgs/#add-or-update-interaction-restrictions-for-an-organization
+func (s *InteractionsService) UpdateRestrictionsForOrg(ctx context.Context, organization string, interaction *InteractionRestriction) (*InteractionRestriction, *Response, error) {
```

Comment:
It just dawned on me that this API currently only takes a single parameter (`limit`) and yet we are passing in a full `InteractionRestriction` struct.

GitHub has recently been clamping down on extra parameters and is wanting to return errors when extra parameters are sent.

Therefore, we have to be extremely careful at what is being sent.

I recommend for now that we pass in another string for the required `limit` parameter and that we document what values can be passed.

Let's also go back and double-check your recent additions for repos and make sure that we are not violating this same principle, as I forgot to check that carefully.

So to be precise, I'm thinking of something like this:

```go
// UpdateRestrictionsForOrg adds or updates the interaction restrictions for an organization.
//
// limit specifies the group of GitHub users who can comment, open issues, or create pull requests
// in public repositories for the given organization.
// Possible values are: "existing_users", "contributors_only", "collaborators_only".
//
// GitHub API docs: https://developer.github.com/v3/interactions/orgs/#add-or-update-interaction-restrictions-for-an-organization
func (s *InteractionsService) UpdateRestrictionsForOrg(ctx context.Context, organization, limit string) (*InteractionRestriction, *Response, error) {
```

And then, within the function, you can still use that same struct, but we are then assured of the other fields being empty:

```go
interaction := &InteractionRestriction{limit: String(limit)}
```

Reply:
> GitHub has recently been clamping down on extra parameters and is wanting to return errors when extra parameters are sent.

Thanks, I didn't know that.

Addressed the review in both organization and repository interactions update API's.